### ISSUE TITLE
passing multiple environment variables

### ISSUE THREAD

**Issue by jairaman12** [hubotio/hubot#1387]
Hello Folks,
My enterprise is looking to provide Hubot to customers as a means to interact in Hipchat and take actions in GitHub, Jenkins & JIRA.
The test setup had a machine user created which connects to all these apps and does the actions that are requested. 
But if handed over to customers, the audit trace-ability might be hampered since all actions will have the machine user auth.
Is there a way to pass multiple customer auth to Hubot?

**Comment by jairaman12** [comment]
Anyone?

**Comment by mose** [comment]
Yeah usually when the bot connects to third party services it will have to use a bot identity, and all actions would be authored by that bot account on github, jira, etc..

If you want the bot to do action on behalf on users, by delegation, you would have to code your own script that will record user api key in the brain, and adapt the code to use user api key when user calls for a command, like adding a comment on a ticket. 

Plugins usually don't do it that way because storing users credentials in the bot requires a safe context and enforced users identity, which with some adapters (like irc) is not enforced by default.
